lib/project_types/theme/commands/share.rb

```diff
@@ -11,8 +11,14 @@ class Share < ShopifyCLI::Command::SubCommand
 
       recommend_default_ruby_range
 
+      options do |parser, flags|
+        parser.on("-f", "--force") { flags[:force] = true }
+      end
+
       def call(_args, name)
         root = root_value(options, name)
+        return unless theme_directory?(root) || current_directory_confirmed?
```

Comment:
I see this logic duplicated across theme commands. Have you considered extracting it into a base class or a module that's included in all the theme classes? We mitigate thus the risk of potential inconsistencies in the future.

Reply:
I have introduced another method so that the lines look now like…
```ruby
return unless valid_theme_directory?(root)
```